tests/checkpointing/test_model_checkpoint.py

```diff
@@ -365,20 +365,24 @@ def test_model_checkpoint_topk_zero(tmpdir):
 def test_model_checkpoint_topk_all(tmpdir):
     """ Test that save_top_k=-1 tracks the best models when monitor key is provided. """
```

Comment:
I was hoping this test could be more explicit, showing that the last is the best and the first is the worst, but I guess it's not so easy with the randomness. 
I will look into this more

Reply:
what if monitor is set to `epoch` with `mode='max'`? this way we can ensure first is the worst and last is the best.

Reply:
But here is we need to track the best model, isn't it? not smallest or largest epoch, I think?

Reply:
yep, it's not ideally correct to consider largest epoch as the best one but since I have set my `monitor='epoch'` with `mode='max'` in such a case checkpoint should monitor `epoch` only and at the end largest epoch will be the best one for the checkpoint_callback.

Reply:
hey, @rohitgr7 I like this idea. This is a valid test because for it to work properly it needs to track the metric correctly, and that is what we want to test! Can we try it?

Reply:
I see, I track smallest `early_stop_on` here, both are valid.
not sure what others think about this?

Reply:
Okay, monitor `epoch` is not in the return metrics

Reply:
I think it's logged by default. If it isn't.. yeah you can subclass `EvalTemplateModel`, override `validation_epoch_end` and log it in the test itself.

Reply:
if it's too many changes, I'd rather keep it for now and follow up with a PR that improves the test. 
I think this PR is high priority to restore master branch from failing.

Reply:
Changed!

Reply:
haha awesome!